lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2769,7 +2817,8 @@ static LogicalResult checkConnectFlow(Operation *connect) {
 
   // TODO: Relax this to allow reads from output ports,
   // instance/memory input ports.
-  if (foldFlow(src) == Flow::Sink) {
+  auto srcFlow = foldFlow(src);
+  if (!isValidSrc(srcFlow)) {
     // A sink that is a port output or instance input used as a source is okay.
     auto kind = getDeclarationKind(src);
```

Comment:
Talked with @rwy7  about this offline: `getDeclarationKind` is not able to walk `ObjectSubfieldOp`s, and will not properly classify declaration kind. (e.g. it may not see that the decl kind is a port).  This limitation is used to produce correct error messages (see test "Try to read from an output object's input ports").